### PR TITLE
CEL-301 Support Httpd Include directives which point to a directory

### PR DESCRIPTION
**Pull request by atopper**
## Description

Touching up original 301 code.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/main/java/com/adobe/aem/dot/dispatcher/core/DispatcherConfigurationFactory.java [hunk 1]
```diff
@@ -60,20 +61,22 @@
...
     File repoFile = new File(repoPath);
-
     if (!repoFile.exists()) {
       throw new ConfigurationException(MessageFormat.format("The `{0}` folder does not exist", repoPath));
     }
 
-    // now we can see if the 'anyDir' exists, this directory contains the dispatcher.any file
-    File dispatcherAnyFile = FileUtils.getFile(repoPath, dispatcherAnyFilePath + File.separator +
-                                                                 DispatcherConstants.DISPATCHER_ANY);
+    File dispatcherAnyFile;
```
Comment (atopper) [inline core/src/main/java/com/adobe/aem/dot/dispatcher/core/DispatcherConfigurationFactory.java:71]: This is the change from last week - do not try to use `dispatcherAnyFilePath` if it is null.  (There was a null in the path.)

File: core/src/main/java/com/adobe/aem/dot/common/util/PathUtil.java [hunk 2]
```diff
@@ -84,6 +84,41 @@ public static String stripLastPathElement(String path) {
...
+    descendantPath = removeDriveLetterPrefix(descendantPath);
+    ancestorPath = removeDriveLetterPrefix(ancestorPath);
+    if (!ancestorPath.endsWith(String.valueOf(File.separatorChar))) {
+      ancestorPath += File.separatorChar;
+    }
+    return descendantPath.startsWith(ancestorPath);
+  }
+
+  /**
+   * Remove
```
Comment (blefebvre) [inline core/src/main/java/com/adobe/aem/dot/common/util/PathUtil.java:103]: Should this be on it's own line like this?

File: core/src/main/java/com/adobe/aem/dot/httpd/core/parser/HttpdConfigurationParser.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public HttpdConfigurationParser(String repoPath) {
     ConfigurationViolations.clearViolations();
 
     List<ConfigurationLine> configLines = this.normalizeConfig(configFile,
-            configFile.getParentFile().getAbsolutePath(), true);
+            configFile.getParentFile().getPath(), true);
```
Comment (blefebvre) [inline core/src/main/java/com/adobe/aem/dot/httpd/core/parser/HttpdConfigurationParser.java:77]: Curious about the reason for this change. Will this affect the file location we report in each violation?

**Review by blefebvre — APPROVED** [review]
Looks good, just a couple of comments.

File: core/src/main/java/com/adobe/aem/dot/httpd/core/parser/HttpdConfigurationParser.java (hunk 3 above)
Comment (atopper) [reply core/src/main/java/com/adobe/aem/dot/httpd/core/parser/HttpdConfigurationParser.java:77]: From what I've read - this is an extreme absolute path.  What the 'real' difference is that it adds the Drive letter on Windows.

**Pull request merged**